src/sources/audiosource.h

```diff
@@ -193,31 +193,66 @@ class AudioSource : public UrlResource, public virtual /*implements*/ IAudioSour
     // Parameters for opening audio sources
     class OpenParams {
       public:
+#ifdef __STEM__
+        OpenParams()
+                : m_signalInfo(), m_stemIdx(mixxx::kNoStemSelectedIdx) {
+        }
+#else
         OpenParams() = default;
+#endif
         OpenParams(
                 audio::ChannelCount channelCount,
-                audio::SampleRate sampleRate)
+                audio::SampleRate sampleRate
+#ifdef __STEM__
+                ,
+                uint stemIdx = mixxx::kNoStemSelectedIdx
```

Comment:
I'm not a big fan of the idea tbh as it would make the NI spec not contained. This means that if in the future we support other stem format, we would need some advanced logic on the "client" side to resolve the supposed stream ID.
Taking your example into account, I think it would make sense to keep `OpenParam` as transparent as possible, without having to be aware of the implementation of the source provider. For example, if the we wanted to add the feature you are suggestion, we could have a `preferredLanguage` that can be consumed by the source provider (in case of MP4, it will check the track label, but in case of other provider it may follow a different logic)

Reply:
> I'm not a big fan of the idea tbh as it would make the NI spec not contained.
I can confirm. My concern is more a general issue. Maybe I have just placed my comment at a wrong position.   

My consideration was sources by looking at: 
`void CachingReaderWorker::newTrack(TrackPointer pTrack, uint stemIdx)`

This exposes the fact that the reader is reading a stem. However the reader can read any stream that is placed in any container. In the same way there can be many streams in an container where only one is the music stream that is usable for DJing. 

The other issue is around the stem and stream = 0 with different meaning, which is confusing in the current state of this PR.  

My idea is to use the stem in the GUI and engine and when reading metadata from the SoundSourceStem, but use a stream index when reading a single stream from a container. In addition we probably need an "autoselect" bzw. "down-mix" value.  

Not sure if this is reasonable. I jsut wanted to explain my concerns. 

The the issue with the same varible name and different interpreatation of zero is the mandatory issue for me.

Reply:
IMHO, `CachingReaderWorker::newTrack` should also explicitly define new tracks settings, even if some of these are redundant (such as stem mask - not idx anymore - on a non-stem file or preferred language on a non-container file), rather than exposing implementation details. Perhaps if more settings were to be added, we could consider wrapping them in a struct instead, something like `NewTrackOpt`? Happy to do that right away.